qt/scientific_interfaces/Indirect/test/IndirectFitDataPresenterTest.h

```diff
@@ -0,0 +1,312 @@
+
+    EXPECT_CALL(*m_model, addWorkspace(workspaceName)).Times(Exactly(1));
+
+    m_view->emitSampleLoaded(QString::fromStdString(workspaceName));
+  }
+
+  void
+  test_that_setSampleWSSuffices_will_set_the_sample_workspace_suffices_in_the_view() {
+    QStringList const suffices{"suffix1", "suffix2"};
+
+    EXPECT_CALL(*m_view, setSampleWSSuffices(suffices)).Times(Exactly(1));
+
+    m_presenter->setSampleWSSuffices(suffices);
+  }
+
+  void
+  test_that_setSampleFBSuffices_will_set_the_sample_file_browser_suffices_in_the_view() {
+    QStringList const suffices{"suffix1", "suffix2"};
+
+    EXPECT_CALL(*m_view, setSampleFBSuffices(suffices)).Times(Exactly(1));
+
+    m_presenter->setSampleFBSuffices(suffices);
+  }
+
+  void
+  test_that_setResolutionWSSuffices_will_set_the_resolution_workspace_suffices_in_the_view() {
+    QStringList const suffices{"suffix1", "suffix2"};
+
+    EXPECT_CALL(*m_view, setResolutionWSSuffices(suffices)).Times(Exactly(1));
+
+    m_presenter->setResolutionWSSuffices(suffices);
+  }
+
+  void
+  test_that_setResolutionFBSuffices_will_set_the_resolution_file_browser_suffices_in_the_view() {
+    QStringList const suffices{"suffix1", "suffix2"};
+
+    EXPECT_CALL(*m_view, setResolutionFBSuffices(suffices)).Times(Exactly(1));
+
+    m_presenter->setResolutionFBSuffices(suffices);
+  }
+
+  void
+  test_that_setStartX_will_alter_the_relevant_startX_column_in_the_data_table() {
+    int const startXColumn(2);
+    TableItem const startX(2.3);
+
+    m_presenter->setStartX(startX.asDouble(), 0, 0);
+
+    assertValueIsGlobal(startXColumn, startX);
+  }
+
+  void
+  test_that_setEndX_will_alter_the_relevant_endX_column_in_the_data_table() {
+    int const endXColumn(3);
+    TableItem const endX(5.5);
+
+    m_presenter->setEndX(endX.asDouble(), 0, 0);
+
+    assertValueIsGlobal(endXColumn, endX);
+  }
+
+  void
+  test_that_the_setExcludeRegion_slot_will_alter_the_relevant_excludeRegion_column_in_the_table() {
+    int const excludeRegionColumn(4);
+    TableItem const excludeRegion("2-3");
+
+    m_presenter->setExclude(excludeRegion.asString(), 0, 0);
+
+    assertValueIsGlobal(excludeRegionColumn, excludeRegion);
+  }
+
+  void test_that_loadSettings_will_read_the_settings_from_the_view() {
+    QSettings settings;
+    settings.beginGroup("/ISettings");
+
+    EXPECT_CALL(*m_view, readSettings(_)).Times(Exactly(1));
+
+    m_presenter->loadSettings(settings);
+  }
+
+private:
+  /// Used in setup
+  void createEmptyTableWidget(int rows, int columns) {
+    m_table = std::make_unique<QTableWidget>(rows, columns);
+    for (auto column = 0; column < rows; ++column)
+      for (auto row = 0; row < columns; ++row)
+        m_table->setItem(row, column, new QTableWidgetItem("item"));
+  }
+
+  void deleteSetup() {
+    m_presenter.reset();
+    m_model.reset();
+    m_view.reset();
+
+    m_dataTablePresenter.reset();
+    m_table.reset();
+  }
+
+  void assertValueIsGlobal(int column, TableItem const &value) const {
```

Comment:
Would this function also be useful in your other set of unit tests that I was reviewing?

Reply:
Yes! I have implemented this on the other set of unit tests in this PR so that I could avoid conflicts. Would it be ok to introduce this function to the other unit tests in this PR?

Reply:
I think that in this instance that is fine, if it makes your life easier!